pypeit/reduce.py

```diff
@@ -681,26 +785,37 @@ def local_skysub_extract(self, waveimg, global_sky, sobjs,
 
         # Could actually create a model anyway here, but probably
         # overkill since nothing is extracted
-
         self.sobjs = sobjs.copy()  # WHY DO WE CREATE A COPY HERE?
         # Loop on slits
         for slit in gdslits:
             msgs.info("Local sky subtraction and extraction for slit: {:d}".format(slit))
             thisobj = (self.sobjs.SLITID == slit) # indices of objects for this slit
+            sobj = self.sobjs[thisobj]
             if np.any(thisobj):
                 thismask = (self.slitmask == slit) # pixels for this slit
                 # True  = Good, False = Bad for inmask
                 inmask = (self.sciImg.mask == 0) & thismask
-                # Local sky subtraction and extraction
-                self.skymodel[thismask], self.objmodel[thismask], self.ivarmodel[thismask], \
-                    self.extractmask[thismask] = skysub.local_skysub_extract(
-                    self.sciImg.image, self.sciImg.ivar, self.tilts, self.waveimg, self.global_sky, self.sciImg.rn2img,
-                    thismask, self.tslits_dict['slit_left'][:,slit], self.tslits_dict['slit_righ'][:, slit],
-                    self.sobjs[thisobj], spat_pix=spat_pix, model_full_slit=self.redux_par['model_full_slit'],
-                    box_rad=self.redux_par['boxcar_radius']/self.spectrograph.detector[self.det-1]['platescale'],
-                    sigrej=self.redux_par['sky_sigrej'],
-                    model_noise=model_noise, std=std, bsp=self.redux_par['bspline_spacing'],
-                    sn_gauss=self.redux_par['sn_gauss'], inmask=inmask, show_profile=show_profile)
+                if self.extraction_par['boxcar_only']:
+                    # TODO -- Make plate_scale an sobj method
+                    if self.spectrograph.pypeline == 'Echelle':
```

Comment:
It looks like we were using a single plate scale for all orders previously which was a mistake. But I don't understand why you are now using the order specific plates scale only for the boxcar only option, but not for the local sky-subtraction which also requires an order specific plate scale. This if echelle should be above before this f self.extraction_par['boxcar_only'].

Reply:
Your comment appears important but is scrambled.

Please paste in the fix to be made (or make it).

Reply:
It looks like we were using a single plate scale for all orders previously which was a mistake. But I don't understand why you are now using the order specific plates scale only for the boxcar only option, but not for the local sky-subtraction which also requires an order specific plate scale. This "if self.spectrograph.pypeline == 'Echelle'" should be above before this "if self.extraction_par['boxcar_only']." , i.e. the order specific plate scale should be used for both types of reductions.

Reply:
Addressed.